src/smt/smt_engine.cpp

```diff
@@ -3235,10 +3273,9 @@ void SmtEnginePrivate::processAssertions() {
                          "Try --bv-div-zero-const to interpret division by zero as a constant.");
   }
 
-  if (options::bitblastMode() == theory::bv::BITBLAST_MODE_EAGER
-      && !options::incrementalSolving())
+  if (options::ackermann() && !options::incrementalSolving())
```

Comment:
The check for whether incremental solving and ackermannization are both enabled is already done earlier, no need to repeat it here.

```suggestion
  if (options::ackermann())
```